### ISSUE TITLE
Randomness

### ISSUE THREAD

**Issue by fcbfcb1998** [liudakai2/UnsupDIS-pytorch#33]
Hello! It is a wonderful work! 
I have an issue. The image alignment training seems to be unrepeatable. The seed was set to 0, but the performance still different every time. Can you show me how to modify the codes?

**Comment by fcbfcb1998** [comment]
I solved this issue.

If you want to have replicable results. you can change init_seed function in general.py to,
def init_seeds(seed=0):
    # Initialize random number generator (RNG) seeds https://pytorch.org/docs/stable/notes/randomness.html
    random.seed(seed)
    np.random.seed(seed)
    torch.manual_seed(seed)
    torch.cuda.manual_seed(seed)
    torch.cuda.manual_seed_all(seed)  # for Multi-GPU, exception safe
    # torch.backends.cudnn.benchmark = True  # AutoBatch problem https://github.com/ultralytics/yolov5/issues/9287
    torch.use_deterministic_algorithms(True)
    torch.backends.cudnn.deterministic = True
    os.environ['CUBLAS_WORKSPACE_CONFIG'] = ':4096:8'
    os.environ['PYTHONHASHSEED'] = str(seed)

**Comment by liudakai2** [comment]
Hi, thanks for your advice. A little randomness is not always a bad thing. But we are happy to provide users with the option of strict reproducibility, as YOLOv5 dose.

**Comment by QinAolong** [comment]
您好，我在windows上进行训练，但是数据集不知道怎么导入，也是采用建立软连接的方式嘛，如果您看到，请您帮我一下，谢谢您

**Comment by liudakai2** [comment]
> 您好，我在windows上进行训练，但是数据集不知道怎么导入，也是采用建立软连接的方式嘛，如果您看到，请您帮我一下，谢谢您

把软连接换成文件夹是一样的

**Comment by QinAolong** [comment]
> > 您好，我在windows上进行训练，但是数据集不知道怎么导入，也是采用建立软连接的方式嘛，如果您看到，请您帮我一下，谢谢您
> 
> 把软连接换成文件夹是一样的

这个怎么换一下，我太菜了，菜哭了

**Comment by liudakai2** [comment]
> > > 您好，我在windows上进行训练，但是数据集不知道怎么导入，也是采用建立软连接的方式嘛，如果您看到，请您帮我一下，谢谢您
> > 
> > 
> > 把软连接换成文件夹是一样的
> 
> 这个怎么换一下，我太菜了，菜哭了

就是直接新建文件夹，不要软连接，然后把数据放在文件夹里面

**Comment by QinAolong** [comment]
> > > > 您好，我在windows上进行训练，但是数据集不知道怎么导入，也是采用建立软连接的方式嘛，如果您看到，请您帮我一下，谢谢您
> > > 
> > > 
> > > 把软连接换成文件夹是一样的
> > 
> > 
> > 这个怎么换一下，我太菜了，菜哭了
> 
> 就是直接新建文件夹，不要软连接，然后把数据放在文件夹里面

那这样的话，是不是把yaml的路径改成我的数据集的路径就好了
![image](https://github.com/liudakai2/UnsupDIS-pytorch/assets/96297179/e7a64c7c-c4d5-4584-aaf2-4f0025fb5460)
